### PR TITLE
INTMDB-572: HELP-42215 - PATCH Update Advanced Configuration Options for One Cluster / Advance Cluster

### PR DESCRIPTION
**Pull request by martinstibbe**
## Description

HELP-42215 - PATCH Update Advanced Configuration Options for One Cluster / Advance Cluster

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_advanced_cluster.go [hunk 1]
```diff
@@ -714,7 +715,7 @@ func resourceMongoDBAtlasAdvancedClusterUpdate(ctx context.Context, d *schema.Re
 	timeout := d.Timeout(schema.TimeoutUpdate)
 
 	// Has changes
-	if !reflect.DeepEqual(cluster, matlas.Cluster{}) {
+	if !reflect.DeepEqual(cluster, clusterChangeDetect) {
```
Comment (Zuhairahmed) [inline mongodbatlas/resource_mongodbatlas_advanced_cluster.go:718]: do we need to make similar change in mongodbatlas_cluster.go as well?

**Review by colm-quinn — APPROVED** [review]
LGTM on the change, potential follow up for @Zuhairahmed question (ok in another PR)

File: mongodbatlas/resource_mongodbatlas_advanced_cluster.go (hunk 1 above)
Comment (martinstibbe) [reply mongodbatlas/resource_mongodbatlas_advanced_cluster.go:718]: Added to cluster resource to avoid extra update call with no changes

**Comment by martinstibbe** [comment]
> LGTM on the change, potential follow up for @Zuhairahmed question (ok in another PR)

I added proper empty struct in cluster one as well to compare so if I change something only in advanced config it does not call update cluster with no changes needed

**Review by colm-quinn — APPROVED** [review]
Thanks for the update, LGTM!

**Review by Zuhairahmed — APPROVED** [review]
Thanks @martinstibbe! LGTM

**Pull request merged**